instaloader/instaloadercontext.py

```diff
@@ -155,6 +157,10 @@ def get_anonymous_session(self) -> requests.Session:
                                 'ig_vw': '1920', 'csrftoken': '',
                                 's_network': '', 'ds_user_id': ''})
         session.headers.update(self._default_http_header(empty_session_only=True))
+        if self.request_timeout is not None:
+            # Override default timeout behavior.
+            # Need to silence mypy bug for this. See: https://github.com/python/mypy/issues/2427
+            session.request = partial(session.request, timeout=self.request_timeout) # type: ignore
```

Comment:
This also needs to be done in copy_session (line 23), load_session_from_file (line 172) and login (line 195), where a requests.Session object is created.